modules/text2vec-octoai/module.go

```diff
@@ -0,0 +1,151 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package modoctoai
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/weaviate/weaviate/modules/text2vec-octoai/ent"
+
+	"github.com/weaviate/weaviate/usecases/modulecomponents/text2vecbase"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/modulecapabilities"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/modules/text2vec-octoai/clients"
+	"github.com/weaviate/weaviate/modules/text2vec-octoai/vectorizer"
+	"github.com/weaviate/weaviate/usecases/modulecomponents/additional"
+)
+
+const Name = "text2vec-octoai"
+
+func New() *OctoAIModule {
+	return &OctoAIModule{}
+}
+
+type OctoAIModule struct {
+	vectorizer                   text2vecbase.TextVectorizerBatch
+	metaProvider                 text2vecbase.MetaProvider
+	graphqlProvider              modulecapabilities.GraphQLArguments
+	searcher                     modulecapabilities.Searcher
+	nearTextTransformer          modulecapabilities.TextTransform
+	logger                       logrus.FieldLogger
+	additionalPropertiesProvider modulecapabilities.AdditionalProperties
+}
+
+func (m *OctoAIModule) Name() string {
+	return Name
+}
+
+func (m *OctoAIModule) Type() modulecapabilities.ModuleType {
+	return modulecapabilities.Text2MultiVec
+}
+
+func (m *OctoAIModule) Init(ctx context.Context,
+	params moduletools.ModuleInitParams,
+) error {
+	m.logger = params.GetLogger()
+
+	if err := m.initVectorizer(ctx, params.GetConfig().ModuleHttpClientTimeout, m.logger); err != nil {
+		return errors.Wrap(err, "init vectorizer")
+	}
+
+	if err := m.initAdditionalPropertiesProvider(); err != nil {
+		return errors.Wrap(err, "init additional properties provider")
+	}
+
+	return nil
+}
+
+func (m *OctoAIModule) InitExtension(modules []modulecapabilities.Module) error {
+	for _, module := range modules {
+		if module.Name() == m.Name() {
+			continue
+		}
+		if arg, ok := module.(modulecapabilities.TextTransformers); ok {
+			if arg != nil && arg.TextTransformers() != nil {
+				m.nearTextTransformer = arg.TextTransformers()["nearText"]
+			}
+		}
+	}
+
+	if err := m.initNearText(); err != nil {
+		return errors.Wrap(err, "init graphql provider")
+	}
+	return nil
+}
+
+func (m *OctoAIModule) initVectorizer(ctx context.Context, timeout time.Duration,
+	logger logrus.FieldLogger,
+) error {
+	octoAIApiKey := os.Getenv("OCTOAI_API_KEY")
```

Comment:
Let's use `OCTOAI_APIKEY` instead of `OCTOAI_API_KEY`

Reply:
ack